Fix actors resupplying even if too far from resupplier

reaperrr opened the pull request:
Fixes #16782.

reaperrr committed:
Fix actors resupplying even if too far from resupplier

pchote commented on OpenRA.Game/Traits/Target.cs:
```diff
@@ -204,6 +204,15 @@ public bool IsInRange(WPos origin, WDist range)
 			return Positions.Any(t => (t - origin).HorizontalLengthSquared <= range.LengthSquared);
 		}
 
+		public bool CenterIsInRange(WPos origin, WDist range)
```
I'm not sure adding this here is the best idea, considering that checking the range at all is a bodge that will be removed when we add proper docking logic. Can we inline the check directly in `Resupply` instead?

reaperrr commented:
Updated.

pchote commented on OpenRA.Mods.Common/Activities/Resupply.cs:
```diff
@@ -79,7 +79,8 @@ public override bool Tick(Actor self)
 			// move actor outside the resupplier footprint.
 			// TODO: This check is nowhere near robust enough, and should be rewritten.
-			if (IsCanceling && host.IsInRange(self.CenterPosition, closeEnough))
+			var isCloseEnough = closeEnough.LengthSquared == 0 || (host.CenterPosition - self.CenterPosition).HorizontalLengthSquared <= closeEnough.LengthSquared;
```
Does anyone know why the original code here used `closeEnough == 0` to mean "disable the range check", rather than `closeEnough < 0`? This removes the ability to say "I want it *exactly* at this position".

reaperrr replied:
I noticed that, too. I think I faintly remember it had something to do with aircraft, but that's from a time well before we started refactoring this. Could very well be obsolete by now.

pchote reviewed (dismissed):
Works as advertised

fruestueck commented:
Tested with RA & TD.

reaperrr committed:
Change Resupply closeEnough 'infinite' to negative

...instead of 'zero'.

Returning 'true' at a distance of zero was a legacy left-over
that isn't used anymore once #16695 is merged.

reaperrr commented:
Updated.

Note that the new commit might regress re-queued aircraft Resupply (on Stop orders), but only until #16695 is merged.

reaperrr commented:
Updated last commits' title and message.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/78d74618eb9b81a396eda367caba510c452d8763...ff4cf7d38d6275e0074064703c2c11040afe3657)
